File: ios/Sources/MsalPluginPlugin/MsalPluginPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(MsalPluginPlugin)
public class MsalPluginPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "MsalPluginPlugin"
    public let jsName = "MsalPlugin"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "echo", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = MsalPlugin()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
}

File: android/src/main/kotlin/com/voortman/capacitor/msal/MsalPluginPlugin.kt
package com.voortman.capacitor.msal

import com.getcapacitor.JSObject
import com.getcapacitor.Plugin
import com.getcapacitor.PluginCall
import com.getcapacitor.annotation.CapacitorPlugin
import com.getcapacitor.PluginMethod

@CapacitorPlugin(name = "MsalPlugin")
class MsalPluginPlugin : Plugin() {

    private val implementation = MsalPlugin()

    @PluginMethod
    fun echo(call: PluginCall) {
        val value = call.getString("value") ?: ""

        val ret = JSObject().apply {
            put("value", implementation.echo(value))
        }
        call.resolve(ret)
    }
}
